add option to mirror ldap groups into netbox

alexhoulton opened the pull request:
## New Behavior

Users authenticating with LDAP will have their assigned groups mirrored into Netbox when the environment variable AUTH_LDAP_MIRROR_GROUPS is True

## Contrast to Current Behavior

No change from current behaviour, as the default is None.

## Discussion: Benefits and Drawbacks

Without the environment variable AUTH_LDAP_MIRROR_GROUPS set to True, groups are not mirrored, and parts of Netbox (such as the API) are unable to make use of LDAP group membership for permissions. This change introduces the ability to set this option when using Netbox through docker, and thus benefits members of the community who require this option.

This is not a breaking changes, and is therefore backwards compatible.

## Proposed Release Note Entry

> ## Add `AUTH_LDAP_MIRROR_GROUPS` configuration #327
> 
> Add ability to mirror ldap groups into Netbox (`AUTH_LDAP_MIRROR_GROUPS`)

## Double Check

* [x] I have read the comments and followed the PR template.
* [x] I have explained my PR according to the information in the comments.
* [x] My PR targets the `develop` branch.

cimnine commented on configuration/ldap_config.py:
```diff
@@ -71,6 +71,7 @@ def import_group_type(group_type_name):
 
 # For more granular permissions, we can map LDAP groups to Django groups.
 AUTH_LDAP_FIND_GROUP_PERMS = os.environ.get('AUTH_LDAP_FIND_GROUP_PERMS', 'True').lower() == 'true'
+AUTH_LDAP_MIRROR_GROUPS = os.environ.get('AUTH_LDAP_MIRROR_GROUPS', '').lower() == 'true'
```
If the default is false, please make this explicit:

```suggestion
AUTH_LDAP_MIRROR_GROUPS = os.environ.get('AUTH_LDAP_MIRROR_GROUPS', 'False').lower() == 'true'
```

alexhoulton replied:
Set the default to None as per https://django-auth-ldap.readthedocs.io/en/latest/reference.html#auth-ldap-mirror-groups

cimnine commented:
Thank you for this PR. Please see my comment. Also please rebase onto the latest `develop` branch. This should fix some of the _required_ tests for your.

cimnine commented:
Also: Sorry for the long wait.

ikaremostest committed:
add option to mirror ldap groups into netbox

this commit allows ldap assigned groups to be mirrored into
netbox users.

The default is None as this is not the primary way to do this
change.
